control-plane/gateways/deployment.go

```diff
@@ -156,11 +153,31 @@ func compareDeployments(a, b *appsv1.Deployment) bool {
 	return *b.Spec.Replicas == *a.Spec.Replicas
 }
 
-func deploymentReplicaCount(deployment *pbmesh.Deployment, currentReplicas *int32) *int32 {
-	// TODO NET-6721 tamp replica count up and down based on min and max values
-	instanceValue := globalDefaultInstances
+func deploymentReplicaCount(replicas *meshv2beta1.GatewayClassReplicasConfig, currentReplicas *int32) *int32 {
+	// if we have the replicas config, use it
+	if replicas != nil && currentReplicas == nil {
+		return replicas.Default
+	}
+
+	// if we have the replicas config and the current replicas, use the min/max to ensure
+	// the current replicas are within the min/max range
+	if replicas != nil && currentReplicas != nil {
+		if *currentReplicas > *replicas.Max {
```

Comment:
Even though we know `replicas` is not `nil`, do we still have a risk of `nil` pointer deref if `replicas.Max` is `nil`?

Reply:
yes, it's possible that a user wants to configure a min but not a max, for example, in which case `replicas.Max` would be nil

Reply:
so we don't give that much control to the user, `replicas` is just a single number https://developer.hashicorp.com/consul/docs/k8s/helm#v-meshgateway-replicas

Reply:
and to see how we handle it: https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/templates/gateway-resources-configmap.yaml#L65, we could do the nil checks here to future proof it though